Harbor upgrade failed while upgrading VIC OVA 1.4.0 to 1.4.1RC2

@lgayatri commented on [Fri Jun 29 2018](https://github.com/vmware/harbor/issues/5218)

@renmaosheng @hickeng @zjs @luguang 

Builds used : 

1.4.0 GA - https://storage.googleapis.com/vic-product-ova-releases/vic-v1.4.0-4944-f168720a.ova
1.4.1 RC2 - https://storage.googleapis.com/vic-product-ova-releases/vic-v1.4.1-rc2-5496-f2fdac75.ova
OVA has been upgraded from 1.4.0 to 1.4.1 using https://vmware.github.io/vic-product/assets/files/html/1.4/vic_vsphere_admin/upgrade_appliance.html 

/var/
[upgrade.log](https://github.com/vmware/harbor/files/2149787/upgrade.log)
log/vmware/upgrade.log is attached
```
...
Getting password from harbor.cfg
2018-06-29 14:21:24 [=] Performing pre-upgrade checks
2018-06-29 14:21:24 [=] Starting Harbor upgrade
2018-06-29 14:21:24 [=] [=] Shutting down Harbor
2018-06-29 14:21:26 [=] [=] Migrating Harbor configuration and data
2018-06-29 14:21:26 [=] Testing database credentials...
Trying to start mysql server...
Waiting for MySQL start...
2018-06-29 14:21:28 140737353451456 [Note] mysqld (mysqld 10.2.14-MariaDB) starting as process 8 ...
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Uses event mutexes
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Compressed tables use zlib 1.2.8
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Number of pools: 1
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Using SSE2 crc32 instructions
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Initializing buffer pool, total size = 256M, instances = 1, chunk size = 128M
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Completed initialization of buffer pool
2018-06-29 14:21:28 140736431363840 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: The first innodb_system data file 'ibdata1' did not exist. A new tablespace will be created!
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Setting file './ibdata1' size to 12 MB. Physically writing the file full; Please wait ...
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: File './ibdata1' size is now 12 MB.
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Setting log file ./ib_logfile101 size to 50331648 bytes
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Setting log file ./ib_logfile1 size to 50331648 bytes
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Renaming log file ./ib_logfile101 to ./ib_logfile0
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: New log files created, LSN=45790
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Doublewrite buffer not found: creating new
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Doublewrite buffer created
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: 128 out of 128 rollback segments are active.
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Creating foreign key constraint system tables.
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Creating tablespace and datafile system tables.
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Creating sys_virtual system tables.
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Creating shared tablespace for temporary tables
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: Waiting for purge to start
2018-06-29 14:21:28 140737353451456 [Note] InnoDB: 5.7.21 started; log sequence number 0
2018-06-29 14:21:28 140737353451456 [Note] Plugin 'FEEDBACK' is disabled.
2018-06-29 14:21:28 140737353451456 [ERROR] Could not open mysql.plugin table. Some plugins may be not loaded
2018-06-29 14:21:28 140737353451456 [ERROR] Can't open and lock privilege tables: Table 'mysql.servers' doesn't exist
2018-06-29 14:21:28 140737353451456 [Note] Server socket created on IP: '::'.
2018-06-29 14:21:28 140737353451456 [ERROR] Fatal error: Can't open and lock privilege tables: Table 'mysql.user' doesn't exist
timeout. Can't run mysql server.
DB test failed.
Fail to validate: please make sure your DB auth is correct.
2018-06-29 14:22:29 [=]
2018-06-29 14:22:29 [=] -------------------------
2018-06-29 14:22:29 [=] Upgrade failed.
2018-06-29 14:22:29 [=] Please save /var/log/vmware/upgrade.log and contact VMware support.
2018-06-29 14:22:29 [=] -------------------------
2018-06-29 14:22:29 [=]
```

--- a/installer/build/scripts/upgrade/upgrade.sh
+++ b/installer/build/scripts/upgrade/upgrade.sh
@@ -348,7 +348,7 @@ function moveDisks {
       govc device.remove -vm="$NEW_VM_NAME" "$disk"
     fi
     # Only remove log and database disks from 1.3.0 and greater
-    if [[ ( "$disk" =~ -2$ || "$disk" =~ -3$ ) && ( "$ver" == "$VER_1_3_0"  ||  "$ver" == "$VER_1_3_1" ) ]]; then
+    if [[ ( "$disk" =~ -2$ || "$disk" =~ -3$ ) && ( "$ver" != "$VER_1_2_1" ) ]]; then
       echo "Remove disk $disk from $NEW_VM_NAME"
       govc device.remove -vm="$NEW_VM_NAME" "$disk"
     fi
@@ -363,7 +363,7 @@ function moveDisks {
     exit 1
   fi
 
-  if [ "$ver" == "$VER_1_3_0" ] || [ "$ver" == "$VER_1_3_1" ]; then
+  if [ "$ver" != "$VER_1_2_1" ]; then
     OLD_DB_DISK=$(govc vm.info -json "$OLD_VM_NAME" | jq -r ".VirtualMachines[].Layout.Disk[2].DiskFile[0]" | awk '{print $NF}')
     echo "OLD_DB_DISK: $OLD_DB_DISK"
     OLD_LOG_DISK=$(govc vm.info -json "$OLD_VM_NAME" | jq -r ".VirtualMachines[].Layout.Disk[3].DiskFile[0]" | awk '{print $NF}')
@@ -377,7 +377,7 @@ function moveDisks {
 
   log "Copying old data disk. Please wait."
   govc datastore.cp -ds "$OLD_DATASTORE" -ds-target "$NEW_DATASTORE" "$OLD_DATA_DISK" "$NEW_DATA_DISK" || ( log "Failed to copy data disk. Please try again. Exiting..." && exit 1)
-  if [ "$ver" == "$VER_1_3_0" ] || [ "$ver" == "$VER_1_3_1" ]; then
+  if [ "$ver" != "$VER_1_2_1" ]; then
     log "Copying old database disk. Please wait."
     govc datastore.cp -ds "$OLD_DATASTORE" -ds-target "$NEW_DATASTORE" "$OLD_DB_DISK" "$NEW_DB_DISK" || ( log "Failed to copy database disk. Please try again. Exiting..." && exit 1)
     log "Copying old log disk. Please wait."
@@ -387,7 +387,7 @@ function moveDisks {
   # TODO rename to new version
   echo "Attaching migrated disks to new VIC appliance"
   govc vm.disk.attach -vm="$NEW_VM_NAME" -ds "$NEW_DATASTORE" -disk "$NEW_DATA_DISK" || (log "Failed to attach data disk" && exit 1)
-  if [ "$ver" == "$VER_1_3_0" ] || [ "$ver" == "$VER_1_3_1" ]; then
+  if [ "$ver" != "$VER_1_2_1" ]; then
     govc vm.disk.attach -vm="$NEW_VM_NAME" -ds "$NEW_DATASTORE" -disk "$NEW_DB_DISK" || (log "Failed to attach database disk"  && exit 1)
     govc vm.disk.attach -vm="$NEW_VM_NAME" -ds "$NEW_DATASTORE" -disk "$NEW_LOG_DISK" || (log "Failed to attach log disk" && exit 1)
   fi

